src/backend/distributed/citus.sql

```diff
@@ -332,8 +341,28 @@ CREATE OR REPLACE FUNCTION citus_drop_trigger()
     SET search_path = pg_catalog
     /* declared as SECURITY DEFINER in upgrade script */
     AS $cdbdt$
-DECLARE v_obj record;
+DECLARE
+	v_obj record;
+	sequence_names text[] := '{}';
+	node_names text[] := '{}';
+	node_ports bigint[] := '{}';
+	node_name text;
+	node_port bigint;
 BEGIN
+	SELECT array_agg(object_identity) INTO sequence_names
+	FROM pg_event_trigger_dropped_objects()
+	WHERE object_type = 'sequence';
+
+	SELECT array_agg(sp.nodename), array_agg(sp.nodeport)
+	INTO node_names, node_ports
+	FROM pg_event_trigger_dropped_objects() AS dobj,
+	     pg_dist_shard AS s,
+	     pg_dist_shard_placement AS sp
+	WHERE dobj.object_type IN ('table', 'foreign table')
+	  AND dobj.objid = s.logicalrelid
+	  AND s.shardid = sp.shardid
+	  AND sp.shardstate = 1;
```

Comment:
Huh. I could have sworn `master_drop_all_shards` used only finalized placements, but apparently not. Removing.